File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int>s;

        for(auto a:tokens){
            if(a=="+"||a=="-"||a=="*"||a=="/"){
                int x,y;
                y=s.top(); s.pop();
                x=s.top(); s.pop();
                if(a=="+") s.push(x+y);
                else if(a=="-") s.push(x-y);
                else if(a=="*") s.push(x*y);
                else if(a=="/") s.push(x/y);
            }
            else s.push(stoi(a));
        }
        return s.top();
        
    }
};

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.java
import java.util.Stack;

class Solution {

    int solve(int b, int a, String op) {
        switch (op) {
            case "+": return b + a;
            case "-": return b - a;
            case "*": return b * a;
            case "/": return b / a;
        }
        return 0;
    }

    public int evalRPN(String[] tokens) {
        Stack<Integer> st = new Stack<>();

        for (String token : tokens) {
            if (token.equals("+") || token.equals("-") ||
                token.equals("*") || token.equals("/")) {

                int a = st.pop();
                int b = st.pop();
                st.push(solve(b, a, token));

            } else {
                st.push(Integer.parseInt(token));
                
            }
        }
        return st.peek();
    }
}
